// plugin-welcome/src/api/models/ControllersSpaciality.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface ControllersSpaciality
 */
export interface ControllersSpaciality {
    /**
     * 
     * @type {number}
     * @memberof ControllersSpaciality
     */
    organdata?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersSpaciality
     */
    physiciandata?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersSpaciality
     */
    typediseasedata?: number;
}

export function ControllersSpacialityFromJSON(json: any): ControllersSpaciality {
    return ControllersSpacialityFromJSONTyped(json, false);
}

export function ControllersSpacialityFromJSONTyped(json: any, ignoreDiscriminator: boolean): ControllersSpaciality {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'organdata': !exists(json, 'organdata') ? undefined : json['organdata'],
        'physiciandata': !exists(json, 'physiciandata') ? undefined : json['physiciandata'],
        'typediseasedata': !exists(json, 'typediseasedata') ? undefined : json['typediseasedata'],
    };
}

export function ControllersSpacialityToJSON(value?: ControllersSpaciality | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'organdata': value.organdata,
        'physiciandata': value.physiciandata,
        'typediseasedata': value.typediseasedata,
    };
}
