// plugin-welcome/src/api/models/EntOrgan.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntOrganEdges,
    EntOrganEdgesFromJSON,
    EntOrganEdgesFromJSONTyped,
    EntOrganEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntOrgan
 */
export interface EntOrgan {
    /**
     * OrganName holds the value of the "OrganName" field.
     * @type {string}
     * @memberof EntOrgan
     */
    organName?: string;
    /**
     * 
     * @type {EntOrganEdges}
     * @memberof EntOrgan
     */
    edges?: EntOrganEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntOrgan
     */
    id?: number;
}

export function EntOrganFromJSON(json: any): EntOrgan {
    return EntOrganFromJSONTyped(json, false);
}

export function EntOrganFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntOrgan {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'organName': !exists(json, 'OrganName') ? undefined : json['OrganName'],
        'edges': !exists(json, 'edges') ? undefined : EntOrganEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntOrganToJSON(value?: EntOrgan | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'OrganName': value.organName,
        'edges': EntOrganEdgesToJSON(value.edges),
        'id': value.id,
    };
}
