// plugin-welcome/src/api/models/EntTypeDisease.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntTypeDiseaseEdges,
    EntTypeDiseaseEdgesFromJSON,
    EntTypeDiseaseEdgesFromJSONTyped,
    EntTypeDiseaseEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntTypeDisease
 */
export interface EntTypeDisease {
    /**
     * DiseaseName holds the value of the "DiseaseName" field.
     * @type {string}
     * @memberof EntTypeDisease
     */
    diseaseName?: string;
    /**
     * 
     * @type {EntTypeDiseaseEdges}
     * @memberof EntTypeDisease
     */
    edges?: EntTypeDiseaseEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntTypeDisease
     */
    id?: number;
}

export function EntTypeDiseaseFromJSON(json: any): EntTypeDisease {
    return EntTypeDiseaseFromJSONTyped(json, false);
}

export function EntTypeDiseaseFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntTypeDisease {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'diseaseName': !exists(json, 'DiseaseName') ? undefined : json['DiseaseName'],
        'edges': !exists(json, 'edges') ? undefined : EntTypeDiseaseEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntTypeDiseaseToJSON(value?: EntTypeDisease | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'DiseaseName': value.diseaseName,
        'edges': EntTypeDiseaseEdgesToJSON(value.edges),
        'id': value.id,
    };
}
